components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -89,8 +102,13 @@ AtmosphereOutput (const ekat::Comm& comm, const ekat::ParameterList& params,
 {
   using vos_t = std::vector<std::string>;
 
-  if (params.isParameter("Fill Value")) {
-    m_fill_value = static_cast<float>(params.get<double>("Fill Value"));
+  if (params.isParameter("fill_value")) {
+    m_fill_value = static_cast<float>(params.get<double>("fill_value"));
+  }
+  if (params.isParameter("fill_threshold")) {
+    m_avg_coeff_threshold = params.get<Real>("fill_threshold");
+  } else {
+    m_avg_coeff_threshold = 0.5; // default to 0.5
   }
```

Comment:
No need for if-else, There is a built in default arg
```
m_avg_coeff_threshold = params.get<Real>("fill_threshold", 0.5);
```

Reply:
That's b/c you are calling `get` on the input parameter list, which is const. The default adds an entry if one is not found, which is why you can't call it on a const list. If you call it on m_params, you should not get the error.

An alternative (which I have thought about a few times) is to change the impl of `get` with a default. The current impl, mimics the behavior of `ParameterList` in Trilinos, but one may argue that it's somewhat of a stealth list modification. Perhaps, when a param is not found, we should simply _return_ the default, without adding an entry to the plist. This would be an ekat mod, but I'm afraid it would not be bwd compatible (which may or may not be a problem, if we agree that the current behavior is not intuitive).

Reply:
ahhh thanks for the explanation.  I'll make the change to use m_params to keep it clean

Reply:
nevermind, this approach didn't work.  So I kept it as is.